overlord/snapstate/snapmgr.go

```diff
@@ -560,6 +604,12 @@ func (m *SnapManager) Ensure() error {
 
 	m.runner.Ensure()
 
+	// needs to run after the runner
```

Comment:
Keep in mind that snap-confine can be invoked before snapd boots. Could we perhaps trigger this from snap-run (on demand)? 

We could do the same for all the "stale" objects:
 - security profiles for snaps
 - launchers (well, only if they run snap-run today)
 - snap-confine's own profile

If we detect that something is not up-to-date we reach out to snapd and call some API like `ensure-ephemeral-state`. Just an idea but I think we could use it to avoid costly stuff from happening on boot.